interface-definitions/include/firewall/gre.xml.i

```diff
@@ -0,0 +1,126 @@
+  </properties>
+  <children>
+    <node name="flags">
+      <properties>
+        <help>GRE flag bits to match</help>
+      </properties>
+      <children>
+        <leafNode name="key">
+          <properties>
+            <help>Header includes optional key field</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <leafNode name="checksum">
+          <properties>
+            <help>Header includes optional checksum</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <leafNode name="sequence">
+          <properties>
+            <help>Header includes a sequence number field</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <node name="not">
+          <properties>
+            <help>Match flags not set</help>
+          </properties>
+          <children>
+            <leafNode name="key">
+              <properties>
+                <help>Header does not include optional key field</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+            <leafNode name="checksum">
+              <properties>
+                <help>Header does not include optional checksum</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+            <leafNode name="sequence">
+              <properties>
+                <help>Header does not include a sequence number field</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+          </children>
+        </node>
+      </children>
+    </node>
+    <leafNode name="inner-proto">
+      <properties>
+        <help>EtherType of encapsulated packet</help>
+        <completionHelp>
+          <list>ip ip6 arp vlan 8021q 8021ad</list>
+        </completionHelp>
+        <valueHelp>
+          <format>u32:0-65535</format>
+          <description>Ethernet protocol number</description>
+        </valueHelp>
+        <valueHelp>
+          <format>ip</format>
+          <description>IPv4</description>
+        </valueHelp>
+        <valueHelp>
+          <format>ip6</format>
+          <description>IPv6</description>
+        </valueHelp>
+        <valueHelp>
+          <format>arp</format>
+          <description>Address Resolution Protocol</description>
+        </valueHelp>
+        <valueHelp>
+          <format>vlan</format>
+          <description>VLAN-tagged Ethernet</description>
+        </valueHelp>
+        <valueHelp>
+          <format>8021q</format>
+          <description>VLAN-tagged Ethernet</description>
+        </valueHelp>
+        <valueHelp>
+          <format>8021ad</format>
+          <description>Bridged Ethernet</description>
+        </valueHelp>
+        <constraint>
+          <validator name="gre-protocol"/>
+        </constraint>
+      </properties>
+    </leafNode>
+    <leafNode name="key">
+      <properties>
+        <help>Tunnel Key</help>
```

Comment:
We don't capitalize every word, unless it's an acronym/abbreviation. "Tunnel key" is just a normal phrase.